### DISCUSSION TITLE
Padosum Wiki

### DISCUSSION THREAD

**Discussion by giscus** [padosum/blog#35] [General]
# Padosum Wiki

![ padosum_wiki.png ]   👋 안녕하세요 최연정의 개인 위키입니다.   📝 [ Today-I-Learned|매일 배운 것을 기록 ]하고 있습니다. 🗂 [ wiki-index ] 📚 [ Book-list ] 💡 …

https://www.padosum.dev/

**Comment by padawanR0k** [comment]
안녕하세요! 최연정님 다름이 아니라 위키형태의 개츠비 테마를 찾다가 흘러들오게된 1인입니다..
다름이 아니라 이 테마의 기존 레포지토리에서는 레이아웃 같은 부분들을 커스텀하는 방법이 나와있지 않더라구요.
그런데 이 블로그 레포지토리를 보니까 `src/gatsby-theme-primer-wiki`를 폴더 만들어서 커스터마이징 하신거 같더라구요..
혹시 이 부분은 이 블로그에서 지원하는 기능이 아닌 gatsby자체 옵션일까요? 이렇게 레이아웃을 커스터마이징하는 방법을 알려면, 구글링 키워드를 어떻게 검색하면 될까요?

(`https://www.gatsbyjs.com/docs/how-to/routing/layout-components/` 이런 부분들을 공부하면될까요..?)

**Reply by padosum** [reply]
안녕하세요 @padawanR0k 님 
말씀하신 것처럼 개츠비에서 테마의 컴포넌트를 수정할 수 있는 "shadowing"이라는 기능을 제공하고 있어요
확인하신 대로 저의 레포지토리처럼 폴더 만들어서 해당 컴포넌트 파일을 만들어 입맛대로 수정하면 그게 기존 테마에 덮어씌워집니다 
[공식 문서](https://www.gatsbyjs.com/docs/how-to/plugins-and-themes/shadowing/)에 해당 내용이 나와있고,
아래 웹페이지들을 참고하시거나 "gatsby theme shadowing"이라고 구글링하시면 될 것 같습니다! 
- https://mesh.dev/front-end-engineering/introduce-web-front-end-team-blog
- https://blog.wsgvet.com/how-to-edit-gatsby-flexiblog-personal-theme/

**Reply by padawanR0k** [reply]
@padosum  와 주말임에도 빠른 답변 감사합니다!!!

**Comment by padawanR0k** [comment]
안녕하세요! 친절하게 알려주신 덕분에 위키를 조금씩 써보고 있습니다 :)
다름이 아니라 저의 경우 빌드하여 올릴때만 모든 포스트의 `최근 수정 시각`의 값이 빌드된 시간으로 변경되고 있는데, 혹시 비슷한 이슈를 경험해보신적 있으실까요? (로컬에서는 이상없음)
+ 빌드와 배포는 cloudflare를 사용중입니다.

**Reply by padosum** [reply]
안녕하세요! 저도 같은 문제를 겪었는데요 ㅎㅎ 
우선 [해당 테마의 코드](https://github.com/theowenyoung/gatsby-theme-primer-wiki/blob/0fa0f85c28ad6ab16837af8bcaa44648c7d2b4e0/theme/gatsby-node.js#L477)를 살펴보시면 `git log`를 이용해 시간을 가져오고 있는걸 확인할 수 있어요 

그래서 배포하기 위해 github에 workflow를 작성하실 때 `actions/checkout`에 옵션을 붙이지 않으면 가장 최근 커밋만 가져오게 됩니다. 그럼 해당 커밋 시간만 조회가 됩니다. `fetch-depth: 0` 을 추가해서 전체 히스토리를 가져와야 해요 
[코드](https://github.com/padosum/blog/blob/39a82000ea8ef55a4ee6051aed4e278f47867428/.github/workflows/preview.yaml#L15)를 참고하시면 될 것 같습니다 

그리고 이거랑 관련해서 조금 정리해둔 문서가 있는데 좀 두서가 없지만.. 도움되시길 바랍니다! 
- https://www.padosum.dev/wiki/Github-Actions-with-Vercel/

**Reply by padawanR0k** [reply]
매번 친절하게 답변해주셔서 정말 감사합니다!! 
덕분에 아예 모르고 있었던 지식을 하나 알아가네요!!
확인해보니 cloudflare pages를 사용하는 경우, 배포할 때 git clone 명령어를 실행할 때 커스텀한 옵션을 전달할 수 없네요 ㅜㅜ
다시 github action을 사용해서 문제 해결했습니다 :)

**Reply by padawanR0k** [reply]
혹시 md파일 상단에 수정시간 입력하는건 obsidian에 확장프로그램을 사용하시는걸까요? 아니면 수동으로 직접 입력하시는건가요?

**Reply by padosum** [reply]
[Update time on edit](https://github.com/beaussan/update-time-on-edit-obsidian)이라는 커뮤니티 플러그인 사용하고 있어요
